crypto/asn1/d2i_pr.c

```diff
@@ -60,18 +61,21 @@ EVP_PKEY *d2i_PrivateKey_ex(int type, EVP_PKEY **a, const unsigned char **pp,
                 goto err;
             EVP_PKEY_free(ret);
             ret = tmp;
+            ERR_pop_to_mark();
             if (EVP_PKEY_type(type) != EVP_PKEY_base_id(ret))
                 goto err;
         } else {
             ASN1err(0, ERR_R_ASN1_LIB);
             goto err;
         }
     }
+    ERR_clear_last_mark();
     *pp = p;
     if (a != NULL)
         (*a) = ret;
     return ret;
  err:
+    ERR_clear_last_mark();
```

Comment:
Similarly here - you can get here with the mark already popped.

Reply:
Thanks for your help and patience!